From 68cc91e4e7b7d30c67fdd8ff96b2fea8b4da0228 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 29 Jul 2020 12:43:41 +0200
Subject: [PATCH 1/1] efi_selftest: block device test requires
 CONFIG_DOS_PARTITION

Do not execute the block device test if CONFIG_DOS_PARTITION=n.

Imply CONFIG_DOS_PARTITION in Kconfig.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/Kconfig  | 2 ++
 lib/efi_selftest/Makefile | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/efi_selftest/Kconfig b/lib/efi_selftest/Kconfig
index 478140330b..ca62436108 100644
--- a/lib/efi_selftest/Kconfig
+++ b/lib/efi_selftest/Kconfig
@@ -1,6 +1,8 @@
 config CMD_BOOTEFI_SELFTEST
 	bool "UEFI unit tests"
 	depends on CMD_BOOTEFI
+	imply PARTITIONS
+	imply DOS_PARTITION
 	imply FAT
 	imply FAT_WRITE
 	imply CMD_POWEROFF if PSCI_RESET || SYSRESET_PSCI
diff --git a/lib/efi_selftest/Makefile b/lib/efi_selftest/Makefile
index 7f849032ed..45ce6859b8 100644
--- a/lib/efi_selftest/Makefile
+++ b/lib/efi_selftest/Makefile
@@ -57,7 +57,7 @@ ifeq ($(CONFIG_GENERATE_ACPI_TABLE),)
 obj-y += efi_selftest_fdt.o
 endif
 
-ifeq ($(CONFIG_BLK)$(CONFIG_PARTITIONS),yy)
+ifeq ($(CONFIG_BLK)$(CONFIG_DOS_PARTITION),yy)
 obj-y += efi_selftest_block_device.o
 endif
 
-- 
2.27.0

